Pms.Services/Pms.Datalayer/Queries/UserPerformanceReviewQuery.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Microsoft.EntityFrameworkCore;
+
+using Pms.Core.Database.Abstraction;
+using Pms.Core.Extensions;
+using Pms.Core.Filtering;
+using Pms.Models;
+using Pms.Shared.Extensions;
+
+namespace Pms.Datalayer.Queries
+{
+    public interface IUserPerformanceReviewQuery : IDbQuery<PmsUserPerformanceReviewDto, UserPerformanceReviewQueryFilter>
+    { }
+
+    public class UserPerformanceReviewQuery(PmsDbContext dbContext) :
+        DbQueryBase<PmsUserPerformanceReviewDto, UserPerformanceReviewQueryFilter>(dbContext),
+        IUserPerformanceReviewQuery
+    {
+        protected override IQueryable<PmsUserPerformanceReviewDto> BuildQuery()
+        {
+            var context = DbContext as PmsDbContext;
+
+            var query = context!.UserPerformanceReviews.AsNoTracking()
+                .ConditionalWhere(() => _criteria.UserId.HasValue,
+                    upr => upr.UserId == _criteria.UserId)
+                .ConditionalWhere(() => _criteria.PerformanceReviewId.HasValue,
+                    upr => upr.PerformanceReviewId == _criteria.PerformanceReviewId)
+                .ConditionalWhereContains(
+                    (() => !string.IsNullOrWhiteSpace(_criteria.CalibrationComments),
+                        _criteria.CalibrationComments!, upr => upr.CalibrationComments))
+                .ConditionalWhere(() => _criteria.EmployeeReviewDate.HasValue,
+                    upr => upr.EmployeeReviewDate == _criteria.EmployeeReviewDate)
+                .ConditionalWhere(() => _criteria.ManagerReviewDate.HasValue,
+                    upr => upr.ManagerReviewDate == _criteria.ManagerReviewDate);
+
+            var queryWithDetails = query
+                .Join(context.Users, upr => upr.UserId, user => user.Id, (upr, user) => new { upr, user })
```

Comment:
This could work, but Entity Framework has already featured prepared for this. Which is Include / IncludeThen. 
You can check the performance review or documentation of Entity Framework about it.